src/Type/Registry/DefaultTypeRegistry.php

```diff
@@ -0,0 +1,233 @@
+<?php declare(strict_types=1);
+
+namespace GraphQL\Type\Registry;
+
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Language\DirectiveLocation;
+use GraphQL\Type\Definition\BooleanType;
+use GraphQL\Type\Definition\Directive;
+use GraphQL\Type\Definition\FloatType;
+use GraphQL\Type\Definition\IDType;
+use GraphQL\Type\Definition\IntType;
+use GraphQL\Type\Definition\ListOfType;
+use GraphQL\Type\Definition\NonNull;
+use GraphQL\Type\Definition\ScalarType;
+use GraphQL\Type\Definition\StringType;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Introspection;
+
+/**
+ * A default implementation of the type registry.
+ *
+ * It will return the standard types by default, but can be initialized with custom types.
+ *
+ * When a type by class is requested, it will return a callable that will instantiate the type on first call.
+ * On subsequent calls, it will return the same instance immediately.
+ */
+class DefaultTypeRegistry implements StandardTypeRegistry
```

Comment:
```suggestion
class DefaultStandardTypeRegistry implements StandardTypeRegistry
```

A bit unwieldy, but I think it is important to differentiate this from userland type registries.

Reply:
I renamed it. If you have a better name, let me know.